presto-native-execution/presto_cpp/main/expression/RowExpressionEvaluator.cpp

```diff
@@ -0,0 +1,419 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/expression/RowExpressionEvaluator.h"
+#include <proxygen/httpserver/ResponseBuilder.h>
+#include "presto_cpp/main/common/Utils.h"
+#include "velox/common/encode/Base64.h"
+#include "velox/exec/ExchangeQueue.h"
+#include "velox/expression/EvalCtx.h"
+#include "velox/expression/Expr.h"
+#include "velox/expression/ExprCompiler.h"
+#include "velox/expression/FieldReference.h"
+
+using namespace facebook::presto;
+using namespace facebook::velox;
+
+namespace facebook::presto::expression {
+
+namespace {
+json toVariableReferenceExpression(const exec::FieldReference* fieldReference) {
```

Comment:
Why is a pointer used instead of a reference. Generally, reference is preferred.